src/JsonService.ts

```diff
@@ -28,28 +28,29 @@ export class JsonService {
     }
 
     public async getJson(url: string, token?: string): Promise<Record<string, unknown>> {
+        const logger = this._logger.create("getJson");
         const headers: HeadersInit = {
             "Accept": this._contentTypes.join(", "),
         };
         if (token) {
-            this._logger.debug("getJson: token passed, setting Authorization header");
+            logger.debug("token passed, setting Authorization header");
             headers["Authorization"] = "Bearer " + token;
         }
 
         let response: Response;
         try {
-            this._logger.debug("getJson, url:", url);
+            logger.debug("url:", url);
             response = await fetch(url, { method: "GET", headers });
         }
         catch (err) {
-            this._logger.error("getJson: network error");
-            throw new Error("Network Error");
+            logger.error("Network Error");
+            throw err;
```

Comment:
is err always from type `Error`?

Reply:
Yeah everything from the DOM should extend Error. Likewise I'm also pretty sure it's also always safe to cast caught errors as `Error` within this project as long as we trust crypto-js and jwt-decode to throw those which I believe they do